src/lib/pickles/test/optional_custom_gates/test_fix_domains.ml

```diff
@@ -167,6 +174,136 @@ let test_fix_domains_with_fixed_lookup_tables () =
+    ; Pickles_types.Plonk_types.Features.
+        { none_bool with foreign_field_add = true; xor = true }
+    ]
+  in
+  assert (
+    List.for_all
+      (fun feature_flags ->
+        let domains =
+          Pickles__Fix_domains.domains ~feature_flags
+            (module Pickles.Impls.Step)
+            etyp_unit etyp_unit main
+        in
+        let log2_size = Pickles_base.Domain.log2_size domains.h in
+        log2_size = exp_output )
+      feature_flags_s )
+
+let test_fix_domains_with_range_check_table () =
+  (* The domain size does not depend on the circuit, only on the feature flags *)
+  let main () =
+    add_plonk_constraint (Raw { kind = Zero; values = [||]; coeffs = [||] })
+  in
+  (* Log2 value. Range check table size is 2^12 *)
+  let exp_output = 13 in
+  (* TODO: use QCheck to generate the different options *)
+  let feature_flags_s =
+    [ Pickles_types.Plonk_types.Features.{ none_bool with range_check0 = true }
+    ; Pickles_types.Plonk_types.Features.{ none_bool with range_check1 = true }
+    ; Pickles_types.Plonk_types.Features.
+        { none_bool with range_check1 = true; foreign_field_add = true }
+    ; Pickles_types.Plonk_types.Features.
+        { none_bool with range_check0 = true; foreign_field_add = true }
+    ; Pickles_types.Plonk_types.Features.
+        { none_bool with range_check0 = true; range_check1 = true }
+    ; Pickles_types.Plonk_types.Features.
+        { none_bool with
+          range_check0 = true
+        ; range_check1 = true
+        ; foreign_field_add = true
+        }
+    ; Pickles_types.Plonk_types.Features.
+        { none_bool with foreign_field_mul = true }
+    ; Pickles_types.Plonk_types.Features.
+        { none_bool with foreign_field_mul = true; foreign_field_add = true }
+    ; Pickles_types.Plonk_types.Features.
+        { none_bool with
+          foreign_field_mul = true
+        ; range_check0 = true
+        ; range_check1 = true
+        }
+    ; Pickles_types.Plonk_types.Features.
+        { none_bool with
+          foreign_field_mul = true
+        ; foreign_field_add = true
+        ; range_check0 = true
+        ; range_check1 = true
+        }
+    ]
+  in
+  assert (
+    List.for_all
+      (fun feature_flags ->
+        let domains =
+          Pickles__Fix_domains.domains ~feature_flags
+            (module Pickles.Impls.Step)
+            etyp_unit etyp_unit main
+        in
+        let log2_size = Pickles_base.Domain.log2_size domains.h in
+        log2_size = exp_output )
+      feature_flags_s )
+
+let test_fix_domains_with_range_check_and_xor_table () =
+  (* The domain size does not depend on the circuit, only on the feature flags *)
+  let main () =
+    add_plonk_constraint (Raw { kind = Zero; values = [||]; coeffs = [||] })
+  in
+  let exp_output = 13 in
+  let feature_flags_s =
+    [ Pickles_types.Plonk_types.Features.
+        { none_bool with range_check0 = true; xor = true }
+    ; Pickles_types.Plonk_types.Features.
```

Comment:
You can have in a circuit just the RangeCheck0, but you cannot have RangeCheck1 alone, without RangeCheck0 as well. Perhaps remove this case?